packages/tools/src/utilities/getCalibratedUnits.ts

```diff
@@ -56,7 +75,154 @@ const getCalibratedAreaUnits = (handles, image): string => {
+ * for converting from internal spacing to image spacing.
+ *
+ * @param handles - to detect if spacing information is different between points
+ * @param image - to extract the calibration from
+ * @returns Object containing the units, area units, and scale
+ */
+const getCalibratedLengthUnitsAndScale = (image, handles) => {
+  const [imageIndex1, imageIndex2] = handles;
+  const { calibration, hasPixelSpacing } = image;
+  let units = hasPixelSpacing ? 'mm' : PIXEL_UNITS;
+  const areaUnits = units + SQUARE;
+  let scale = 1;
+  let calibrationType = '';
+
+  if (
+    !calibration ||
+    (!calibration.type && !calibration.sequenceOfUltrasoundRegions)
+  ) {
+    return { units, areaUnits, scale };
+  }
+
+  if (calibration.type === CalibrationTypes.UNCALIBRATED) {
+    return { units: PIXEL_UNITS, areaUnits: PIXEL_UNITS + SQUARE, scale };
+  }
+
+  if (calibration.sequenceOfUltrasoundRegions) {
+    const supportedRegionsMetadata =
+      calibration.sequenceOfUltrasoundRegions.filter(
+        (region) =>
+          SUPPORTED_REGION_DATA_TYPES.includes(region.regionDataType) &&
+          SUPPORTED_LENGTH_VARIANT.includes(
+            `${region.physicalUnitXDirection},${region.physicalUnitYDirection}`
+          )
+      );
+
+    if (!supportedRegionsMetadata.length) {
+      return { units, areaUnits, scale };
+    }
+
+    const region = supportedRegionsMetadata.find(
+      (region) =>
+        imageIndex1[0] >= region.regionLocationMinX0 &&
+        imageIndex1[0] <= region.regionLocationMaxX1 &&
+        imageIndex1[1] >= region.regionLocationMinY0 &&
+        imageIndex1[1] <= region.regionLocationMaxY1 &&
+        imageIndex2[0] >= region.regionLocationMinX0 &&
+        imageIndex2[0] <= region.regionLocationMaxX1 &&
+        imageIndex2[1] >= region.regionLocationMinY0 &&
+        imageIndex2[1] <= region.regionLocationMaxY1
+    );
+
+    if (region) {
+      scale = 1 / (region.physicalDeltaX * region.physicalDeltaY * 100);
+      calibrationType = 'US Region';
+      units = 'mm';
+    }
+  } else if (calibration.scale) {
+    scale = calibration.scale;
+  }
+
+  return {
+    units: units + (calibrationType ? ` ${calibrationType}` : ''),
+    areaUnits: areaUnits + (calibrationType ? ` ${calibrationType}` : ''),
+    scale,
+  };
+};
+
+const getCalibratedProbeUnitsAndValue = (image, handles) => {
+  const [imageIndex] = handles;
+  const { calibration } = image;
+  let units = ['raw'];
+  let values = [null];
+  let calibrationType = '';
+
+  if (
+    !calibration ||
+    (!calibration.type && !calibration.sequenceOfUltrasoundRegions)
+  ) {
+    return { units, values };
+    // Todo: add support for other scenarios
+  }
+
+  if (calibration.sequenceOfUltrasoundRegions) {
+    // for Probe tool
+    const supportedRegionsMetadata =
+      calibration.sequenceOfUltrasoundRegions.filter(
+        (region) =>
+          SUPPORTED_REGION_DATA_TYPES.includes(region.regionDataType) &&
+          SUPPORTED_PROBE_VARIANT.includes(
```

Comment:
There just aren't that many units to test - is there any reason to just not include all of them, separately in the x and y?  It isn't as though the behaviour changes other than labelling.